### PR TITLE
add PVR.TimeShiftSeekbar infolabel

### PR DESCRIPTION
**Pull request by ronie**
## Description
currently it's not possible for skins to use the timeshift seek slider (POPUP_SEEK_TIMESHIFT_PROGRESS) outside of the SeekBar dialog. it is a built-in control and it's value is set directly by Kodi.

with the 'normal' seek slider (POPUP_SEEK_PROGRESS), it's value is also available to skins though an infolabel `$INFO[Player.Seekbar]`

this PR adds an infolabel `$INFO[PVR.TimeShiftSeekbar]` for the timeshift seek slider as well.

@ksooo: this is a bit of a code refactor, based on how it is currently implemented for the `Player.Seekbar` infolabel.
please check if i didn't mess up anything.

## Motivation and Context
i'll be needing this for https://github.com/xbmc/xbmc/pull/18533

## How Has This Been Tested?
tested in Estuary using a slider control outside of the SeekBar dialog.
also verfied it doesn't break the built-in seek slider.

## Screenshots
seekslider

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp [hunk 1]
```diff
@@ -1764,3 +1767,26 @@ void CPVRGUIInfo::UpdateNextTimer()
   m_tvTimersInfo.UpdateNextTimer();
   m_radioTimersInfo.UpdateNextTimer();
 }
+
+int CPVRGUIInfo::GetTimeShiftSeekPercent() const
+{
+  int progress = 0;
+  progress = m_timesInfo.GetTimeshiftProgressPlayPosition();
```
Comment (ksooo) [inline xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp:1774]: can be on one line `int progress = m_timeIn...`

File: xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp [hunk 2]
```diff
@@ -1764,3 +1767,26 @@ void CPVRGUIInfo::UpdateNextTimer()
...
+
+  int seekSize = g_application.GetAppPlayer().GetSeekHandler().GetSeekSize();
+  if (seekSize != 0)
+  {
+    int total = 0;
+    total = m_timesInfo.GetTimeshiftProgressDuration();
```
Comment (ksooo) [inline xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp:1780]: can be on one line `int totel = m_timesIn...`

File: xbmc/guilib/guiinfo/GUIInfoLabels.h [hunk 3]
```diff
@@ -643,6 +643,7 @@
 #define PVR_TIMESHIFT_PROGRESS_START_TIME (PVR_STRINGS_START + 70)
 #define PVR_TIMESHIFT_PROGRESS_END_TIME   (PVR_STRINGS_START + 71)
 #define PVR_EPG_EVENT_ICON                (PVR_STRINGS_START + 72)
+#define PVR_TIMESHIFT_SEEKBAR             (PVR_STRINGS_START + 73)
 #define PVR_STRINGS_END                   PVR_EPG_EVENT_ICON
```
Comment (ksooo) [inline xbmc/guilib/guiinfo/GUIInfoLabels.h:647]: '#define PVR_STRINGS_END PVR_TIMESHIFT_SEEKBAR' as the new value is now the last string label.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/guilib/guiinfo/GUIInfoLabels.h (hunk 3 above)
Comment (ksooo) [reply xbmc/guilib/guiinfo/GUIInfoLabels.h:647]: Wait, the new label is an int, thus it belongs elsewhere in this file...

Comment (ksooo) [reply xbmc/guilib/guiinfo/GUIInfoLabels.h:647]: Ah, now I have it. Those values (for any reason - was before my time) are both labels and ints. So, please add implementation for PVR_TIMESHIFT_SEEKBAR  also to CPVRGUIInfo::GetPVRLabel

Comment (ronie) [reply xbmc/guilib/guiinfo/GUIInfoLabels.h]: some are label only, some are int only and some are both. it depends on the usecase.
the label ones can be displayed as text in the skin, the int ones are for use in a progress and/or slider control.
the ones that are a dual use, for instance the Volume infolabel can be shown as text as well a in a volume-bar.

Comment (ksooo) [reply xbmc/guilib/guiinfo/GUIInfoLabels.h]: Okay, thanks. This PVR_STRINGS_END things needs to be cleanuo up some day (not in this PR). I find it confusing to name items as "strings" if they aren't. OTOH, PVR_STRINGS_END seems not to be used anywhere.

**Review by ksooo — APPROVED** [review]
lgtm

**Pull request merged**